docs/topics/build-tools-api.md

```diff
@@ -0,0 +1,111 @@
+[//]: # (title: Build tools API)
+
+<primary-label ref="experimental-general"/>
+
+<!-- Add diagram -->
+
+Kotlin 2.2.0 introduces an experimental Build tools API (BTA) that simplifies how build systems integrate with the 
+Kotlin compiler.
+
+Traditionally, adding full Kotlin support to a build system (like incremental compilation, Kotlin compiler plugins, 
+daemons, and Kotlin Multiplatform) required significant effort. The BTA aims to reduce this complexity by providing
+a unified API between build systems and the Kotlin compiler ecosystem.
+
+The BTA defines a single entry point that build systems can implement. This removes the need to deeply integrate with internal compiler details.
+
+Currently, the BTA supports Kotlin/JVM **only**.
+
+> The BTA itself is not yet publicly available for direct use in your own build tool integrations.
+> If you're interested in the proposal or want to share feedback, see the [KEEP](https://github.com/Kotlin/KEEP/issues/421).
+> Follow the status of its implementation in [YouTrack](https://youtrack.jetbrains.com/issue/KT-76255).
+> 
+{style="warning"}
+
+## Integration with Gradle
+
+<primary-label ref="experimental-general"/>
+
+The Kotlin Gradle plugin (KGP) already supports the BTA.
+
+### How to enable
+
+Add the following property to your `gradle.properties` file:
+
+```kotlin
+kotlin.compiler.runViaBuildToolsApi=true
+```
+
+### Configure different compiler versions
+
+With the BTA, you can now use a different Kotlin compiler version than the version used by the KGP. This is useful when:
+
+* You want to try new Kotlin features but haven't updated your build scripts yet.
+* You need the latest plugin fixes but want to stay on an older compiler version for now.
+
+Here's an example of how to configure this in your `build.gradle.kts` file:
+
+```kotlin
+import org.jetbrains.kotlin.buildtools.api.ExperimentalBuildToolsApi
+import org.jetbrains.kotlin.gradle.ExperimentalKotlinGradlePluginApi
+
+plugins {
+	kotlin("jvm") version "2.2.0"
+}
+
+group = "org.jetbrains.example"
+version = "1.0-SNAPSHOT"
+
+repositories {
+	mavenCentral()
+}
+
+kotlin {
+	jvmToolchain(8)
+	@OptIn(ExperimentalBuildToolsApi::class, ExperimentalKotlinGradlePluginApi::class)
+	compilerVersion.set("2.1.21") // <-- different version than 2.2.0
+}
+```
+
+#### Compatible Kotlin compiler and KGP versions
+
+The BTA supports:
+
+* The three previous major Kotlin compiler versions.
+* One major version forward.
+
+For example, in KGP 2.2.0, the supported Kotlin compiler versions are:
```

Comment:
I think it's the 1st time that I see KGP having it's own version written like that. Plus, the intro doesn't correspond well with the last item on the list. consider this:

> For example, in Kotlin 2.2.0, the BTA is compatible with (the following compiler versions):

Reply:
We have the KGP version like that in our compatibility table: https://kotlinlang.org/docs/gradle-configure-project.html#apply-the-plugin 🤔 

The Kotlin Gradle plugin and the Kotlin compiler can have different versions so Kotlin 2.2.0 alone is unclear. I don't think we should describe the BTA as the actor here because it enables the fact that the Kotlin compiler version can be different from the KGP version. I could separate the last bullet to become two separate bullets for each version, if you think that makes more sense?